algorithms/src/std_algorithms/impl/Kokkos_ExclusiveScan.hpp

```diff
@@ -96,6 +96,8 @@ struct ExclusiveScanDefaultFunctor {
 
   KOKKOS_FUNCTION
   void join(value_type& update, const value_type& input) const {
+    if (input.is_initial) return;
+
```

Comment:
@fnrizzi With a simplified `parallel_scan` implementation I noticed that
```
reducer.init(value2);
reducer.join(value1, value2);
```
would give wrong results (`value1` shouldn't change but does) as opposed to
```
reducer.init(value2);
reducer.join(value2, value1);
```
where we get `value2==value1` as expected (assuming `value1` doesn't have initial value).

Reply:
I am not sure i am following what is going on here... did you try to create a reducer and call init and joint as above? 

Btw, why for `reducer.join(value1, value2);` you say that: "would give wrong results (value1 shouldn't change but does". the first parameter in join is what changes , so value1 changes

Reply:
Yes, I was running into a situation where `update` had a non-initial value and `input` had an initial value (representing the neutral element). This gave me the wrong results when testing an inclusive scan with a multiplicative functor since the original value was multiplied by zero.